### PR TITLE
Application system - bootstrap data protection complaint

### PR DESCRIPTION
**Pull request by atligudl**
# \<Application system - bootstrap data protection complaint\>

## What

Bootstrap Data Protection Complaint
Set up drafts for first screens and components

## Why

Prepare for development of data protection authority complaint form

## Screenshots / Gifs

2021-03-04 15_10_17-Window

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/data-protection-complaint/src/lib/dataSchema.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import { error } from './messages/error'
+
+export enum OnBehalf {
+  MYSELF = 'myself',
+  MYSELF_AND_OR_OTHERS = 'myselfAndOrOthers',
+  COMPANY = 'company',
+  ORGANIZATION_OR_INSTITUTION = 'organizationOrInsititution',
+}
+
+const nationalIdRegex = /([0-9]){6}-?([0-9]){4}/
```
Comment (darrikonn) [inline libs/application/templates/data-protection-complaint/src/lib/dataSchema.ts:12]: use the kennitala package (this regex isn't correct for national Ids):
`import * as kennitala from 'kennitala'`

**Review by darrikonn — COMMENTED** [review]
nice

File: libs/application/templates/data-protection-complaint/src/lib/dataSchema.ts (hunk 1 above)
Comment (darrikonn) [reply libs/application/templates/data-protection-complaint/src/lib/dataSchema.ts:12]: e.g. `1234567890` would pass, but isn't valid.

Comment (atligudl) [reply libs/application/templates/data-protection-complaint/src/lib/dataSchema.ts]: Sounds good, we might wanna inform the other teams about this too

File: workspace.json [hunk 2]
```diff
@@ -1525,6 +1525,35 @@
...
+          }
+        },
+        "test": {
+          "builder": "@nrwl/jest:jest",
+          "options": {
+            "jestConfig": "libs/application/templates/data-protection-complaint/jest.config.js",
+            "passWithNoTests": true
+          }
+        }
+      }
```
Comment (jeremybarbet) [inline workspace.json:1555]: You might want to add the extract-script to handle translation here https://docs.devland.is/handbook/libs/localization#message-extraction

**Review by jeremybarbet — COMMENTED** [review]
Just a minor comment, the rest looks good to me

**Review by jeremybarbet — DISMISSED** [review]
LGTM

File: libs/application/templates/data-protection-complaint/.eslintrc [hunk 3]
```diff
@@ -0,0 +1,249 @@
+{
```
Comment (sindrig) [inline libs/application/templates/data-protection-complaint/.eslintrc:1]: Probably best if you use extends here like most of the other apps/libs?

```
{ "extends": "../../.eslintrc", "rules": {}, "ignorePatterns": ["!**/*"] }
```

Comment (atligudl) [reply libs/application/templates/data-protection-complaint/.eslintrc:1]: Sure thing, we might wanna do that for the other templates as well then

**Review by jeremybarbet — APPROVED** [review]

**Review by sindrig — APPROVED** [review]
LGTM

**Pull request merged**